### PR TITLE
Database changes

### LINKED ISSUE #31: Change Tracks to Competitions
Rename tracks table to competitions; add description, genre, and rules columns

Will also need to:
1. change references in verses table to match
2. change belongs_to relation in Verse model
3. change upload endpoint creation to Competition instead of track
4. change factory definition to match
5. Rename Track model to Competition model
6. Update api contract
7. Profit?

### PR DESCRIPTION
**Pull request by the-color-bliu**
## Made changes to database
Add column title to verses, change tracks table to competitions, add genre, description, and rules column 

This PR closes #31 #36 

- [x] Bug fix (non-breaking change which fixes an issue): 

- [ ] New feature (non-breaking change which adds functionality):

- [ ] Breaking change (fix or feature that would cause existing functionality to change):

- [x] My change requires a change to the documentation.

- [ ] I have updated the documentation accordingly.

- [ ] I have added tests to cover my changes.

- [x] All new and existing tests passed.

### CODE REVIEW

File: app/controllers/api/v1/upload_controller.rb [hunk 1]
```diff
@@ -11,10 +11,14 @@ def create
...
+        #   audio_path: key,
+        #   title: params[:title],
+        #   user_id: params[:userId],
+        #   track_id: params[:trackId]
+        # )
         verse = Verse.create(
-          audio_path: key,
-          user_id: params[:userId],
-          track_id: params[:trackId]
+          verse_params.merge({audio_path: key})
```
Comment (ETBassist) [inline app/controllers/api/v1/upload_controller.rb:21]: This is a pretty easy refactor that should be useful in the future; definitely want to remember to go back over this and change to use this syntax anywhere we're creating a new resource in this controller

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,8 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+user = User.create(
+  name: 'Biz Markie',
+  email: 'justafriend@example.com'
+)
```
Comment (ETBassist) [inline db/seeds.rb:12]: *Clearly* the GOAT

**Review by ETBassist — APPROVED** [review]
Looks Good To me

**Review by SageOfCode — APPROVED** [review]
LGTM!

**Pull request merged**